### PR TITLE
Feature: Timestamp support for JSONSchema generated schemas

### PR DESCRIPTION
**Pull request by noxify**
Problem
=======

Closes: #93 

Solution
========

Add checks for `format == 'date-time`` inside the string type check for`string` and `array of string`

Change summary:
---------------
* Added format check for string fields ( and string arrays )
* Check for JSONSchema property `format` with value `date-time`
* Updated jsonschema tests and updated the snapshots

Steps to Verify:
----------------
1. Generate a JSON Schema or use an existing one
2. Add `"format":"date-time"` to the field which should have a Date/Time value
3. Ensure that the value is a valid `Date` object
4. Enjoy

### CODE REVIEW

**Comment by noxify** [comment]
The parquet schema is the following:

```
❯ parquet-tools schema json-schema-test-file.parquet
message root {
  optional binary string_field (STRING);
  optional int64 int_field;
  optional group array_field (LIST) {
    repeated group list {
      optional binary element (STRING);
    }
  }
  optional group timestamp_array_field (LIST) {
    repeated group list {
      optional int64 element (TIMESTAMP(MILLIS,true));
    }
  }
  optional int64 timestamp_field (TIMESTAMP(MILLIS,true));
  optional group obj_field {
    optional binary sub1 (STRING);
    optional binary sub2 (STRING);
  }
  optional group struct_field (LIST) {
    repeated group list {
      required group element {
        optional binary sub3 (STRING);
        optional binary sub4 (STRING);
        optional group sub5 {
          optional binary sub6 (STRING);
          optional binary sub7 (STRING);
        }
        optional group sub8 (LIST) {
          repeated group list {
            optional binary element (STRING);
          }
        }
      }
    }
  }
}❯ parquet-tools schema json-schema-test-file.parquet
message root {
  optional binary string_field (STRING);
  optional int64 int_field;
  optional group array_field (LIST) {
    repeated group list {
      optional binary element (STRING);
    }
  }
  optional group timestamp_array_field (LIST) {
    repeated group list {
      optional int64 element (TIMESTAMP(MILLIS,true));
    }
  }
  optional int64 timestamp_field (TIMESTAMP(MILLIS,true));
  optional group obj_field {
    optional binary sub1 (STRING);
    optional binary sub2 (STRING);
  }
  optional group struct_field (LIST) {
    repeated group list {
      required group element {
        optional binary sub3 (STRING);
        optional binary sub4 (STRING);
        optional group sub5 {
          optional binary sub6 (STRING);
          optional binary sub7 (STRING);
        }
        optional group sub8 (LIST) {
          repeated group list {
            optional binary element (STRING);
          }
        }
      }
    }
  }
}
```

File: lib/jsonSchema.ts [hunk 1]
```diff
@@ -1,127 +1,159 @@
...
-        case 'boolean':
-            return fields.createListField('BOOLEAN', optionalFieldList);
-        case 'object':
-            return fields.createStructListField(fromJsonSchema(fieldValue.items), optionalFieldList);
-        default:
-            throw new UnsupportedJsonSchemaError(`Array field type ${JSON.stringify(fieldValue.items)} is unsupported.`);
-    }
+  switch (fieldValue.items.type) {
+    case "string":
+      if (fieldValue.items.format && fieldValue.items.format == "date-time") {
```
Comment (wilwade) [inline lib/jsonSchema.ts:84]: Not a change at all, but I cannot wait for https://github.com/tc39/proposal-pattern-matching

File: lib/jsonSchema.ts [hunk 2]
```diff
@@ -1,127 +1,159 @@
...
-    )) {
-        schema[fieldName] = fromField(fieldName, fieldValue);
-    }
+  for (const [fieldName, fieldValue] of Object.entries(
+    jsonSchema.properties || {}
+  )) {
+    schema[fieldName] = fromField(fieldName, fieldValue)
+  }
 
-    return schema;
```
Comment (wilwade) [inline lib/jsonSchema.ts:126]: Also just noting that we will eventually get a linter and formatter, but not today.
https://github.com/LibertyDSNP/parquetjs/issues/34

**Review by wilwade — APPROVED** [review]
Looks good to me and works in a script!

**Review by enddynayn — APPROVED** [review]
👍 awesome!

**Pull request merged**